### PR TITLE
feature(cron): allows for a more systematic way of calling cron using…

### PR DESCRIPTION
**Pull request by wadebenson**
… one url

one cron url to call. http://YOUR.SITE/cron/run/
Where all the cron jobs will execute at the specified internval
when it's time window deadline has been reached. (i.e. 1min, 5min, 15min, ...)

### CODE REVIEW

File: engine/lib/cron.php [hunk 1]
```diff
@@ -35,6 +35,58 @@ function _elgg_cron_init() {
...
+	ob_start();
+
+	$periods = array(
+		'minute' => 60,
+		'fiveminute' => 300,
+		'fifteenmin' => 900,
+		'halfhour' => 1800,
+		'hourly' => 3600,
+		'daily' => 86400,
+		'weekly'=> 604800,
```
Comment (juho-jaakkola) [inline engine/lib/cron.php]: The code analysis tests are failing because there should be a space between ```'weekly'``` and ```=>```.

**Comment by sh3llc0de** [comment]
Ok fixed the space.. sorry for the trouble... I code everything in vi and didn't see it.

**Comment by juho-jaakkola** [comment]
Everything seems to be OK. I'll just let one more Elgg core team member to do a review before merging this.

**Comment by mrclay** [comment]
I'd prefer to wait until 3.0. Plugins should not be depending on URL or the fact that (currently) each queue gets its own request, but some may be. Sites already set up with the old URLs wouldn't be a problem, but new sites on 2.1 would be the concern.

**Comment by sh3llc0de** [comment]
This allows the old and new methods to operate interchangeably.. There is no requirement to use /run/ but allows a user to simplify there cron jobs down to 1.

**Comment by beck24** [comment]
Looks fine to me as long as we also keep the old method.  My concern is that there is a lot of heavy lifting happening on cron, so when you hit monthly cron with this method you're going to actually be doing monthly, weekly, daily, hourly, halfhour, fifteenminute, fiveminute, minute all in the same thread which will likely cause problems in the kind of sites I deal with.

**Comment by mrclay** [comment]
To be sure, this is how it should've been designed originally. But for BC I don't think we should recommend site owners switch to the /cron/run URL until 3.0. *But* I'm OK adding the code to support it in 2.x.

Concerns: Although they *shouldn't*, a plugin may act based on `current_page_url()` or a `'route', 'cron'` hook handler, or replace the "cron" page handler. Any of these would break something if a site owner used that plugin on a new 2.1 site using only /cron/run.

Another less likely: A broken plugin could crash one of the cron intervals, but with /cron/run that crash that would prevent *all* other jobs on that thread from running.

**Comment by mrclay** [comment]
As this doesn't promote /cron/run in the docs, LGTM.

**Comment by juho-jaakkola** [comment]
Thanks for contributing @sh3llc0de!

**Pull request merged**